lib/vault.ts

```diff
@@ -163,6 +191,105 @@ export async function listAllSubmissions(
   }
 }
 
+export async function listAllSubmissionsAndConfirmations(
+  ability: UserAbility,
+  formID: string
+): Promise<{ submissions: VaultSubmissionAndConfirmationList[] }> {
+  try {
+    await checkAbilityToAccessSubmissions(ability, formID);
+  } catch (e) {
+    if (e instanceof AccessControlError)
+      logEvent(
+        ability.userID,
+        { type: "Response" },
+        "AccessDenied",
+        `Attempted to access all submissions and confirmations for form ${formID}`
+      );
+    throw e;
+  }
+
+  try {
+    const documentClient = connectToDynamo();
+
+    let accumulatedResponses: VaultSubmissionAndConfirmationList[] = [];
+    let lastEvaluatedKey = null;
+
+    while (lastEvaluatedKey !== undefined) {
+      const getItemsDbParams: QueryCommandInput = {
+        TableName: "Vault",
+        // Limit the amount of response to 500.  This can be changed in the future once we have pagination.
+        Limit: 500 - accumulatedResponses.length,
+        ExclusiveStartKey: lastEvaluatedKey ?? undefined,
+        KeyConditionExpression: "FormID = :formID and begins_with(NAME_OR_CONF, :namePrefix)",
+        ExpressionAttributeValues: {
+          ":formID": formID,
+          ":namePrefix": "NAME#",
+        },
+        ExpressionAttributeNames: {
+          "#status": "Status",
+          "#name": "Name",
+        },
+        ProjectionExpression:
+          // "FormID,#status,SecurityAttribute,#name,CreatedAt,LastDownloadedBy,ConfirmTimestamp,DownloadedAt,RemovalDate",
+          "FormID,#status,#name,ConfirmationCode",
+      };
+      const queryCommand = new QueryCommand(getItemsDbParams);
+      // eslint-disable-next-line no-await-in-loop
+      const response = await documentClient.send(queryCommand);
+
+      if (response.Items?.length) {
```

Comment:
This parsing function will need to be modified to only address the information you're returning from the dynamodb call. (FormID, Status, Name, ConfirmationCode)